K8SPXC-342 haproxy doc (#478)
percona/percona-xtradb-cluster-operator: source/proxysql-conf.rst

.. proxysql-conf::

Configuring Load Balancing with ProxySQL
========================================

> Percona XtraDB Cluster Operator provides choice of two cluster components to

Reviewer: "provides a choice of"

> carry on load balancing and proxy service: you can use either `HAProxy<https://haproxy.org>`_ or `ProxySQL <https://proxysql.com/>`_.

Reviewer: "provide a load balancing and proxy service"

> You can control which one to use, if any, enabling or disabling via the

Reviewer: "if any, by enabling"

``haproxy.enabled`` and ``proxysql.enabled`` options in the ``deploy/cr.yaml``
configuration file. 

> Use the following command to enable HAProxy:

Reviewer 2: ```suggestion
Use the following command to enable ProxySQL:
```

.. code:: bash

>    kubectl patch pxc cluster1 --type=merge --patch '{ \
>     "spec": {"haproxy":{ "enabled": false }, \
>     "proxysql":{ "enabled": true } \
>     }}'

Reviewer 2: please check command with @tplavcic 
```suggestion
   kubectl patch pxc cluster1 --type=merge --patch '{ "spec": {
    "haproxy":{ "enabled": false } 
    "proxysql":{ "enabled": true }, 
    "proxysql":{ "size": 3 },
    "proxysql":{ "image": "percona/percona-xtradb-cluster-operator:1.5.0-proxysql" }
    }}'
```

.. note:: For obvious reasons the Operator will not allow the simultaneous
   enabling of both HAProxy and ProxySQL.

> The resulting setup will use zero PXC node (``cluster1-pxc-0`` by default)

Reviewer: "will use the number zero PXC node"

Reviewer 2: also please `s/node/member/` in all docs

as writer.

> When the cluster with ProxySQL is upgraded, the following steps

Reviewer: "When a cluster with ProxySQL"

> When the cluster with ProxySQL is upgraded, the following steps
> take place. First, reader nodes are upgraded one by one: the Operator waits
> until the upgraded node shows up in ProxySQL with online status, and then
> proceeds to upgrade the next node. When the upgrade is finished for all reader
> nodes, then writer PXC node is finally upgraded.

Reviewer 2: please `s/node/PXC member/` in all documents.
reason: when we say "node" -- user can think about K8s node

```suggestion
When the cluster with ProxySQL is upgraded, the following steps
take place. First, reader memebers are upgraded one by one: the Operator waits
until the upgraded node shows up in ProxySQL with online status, and then
proceeds to upgrade the next memeber. When the upgrade is finished for all reader
memeber, then the writer PXC member is finally upgraded.
```
